### ISSUE TITLE
::selection not checked for color contrast

### ISSUE THREAD

**Issue by AutoSponge** [dequelabs/axe-core#1926]
Expectation: Overriding `::selection` with a poor contrast ratio should cause a violation when text is selected.

Actual: no violation.

Motivation: This should be something axe can find given the state of a page.

<pre><code>
axe-core version: 3.4.0
axe-webdriver, extension or other integration version: X.Y.Z

Browser and Assistive Technology versions
</code></pre>

Example:
```html
    <section class="selection">
      <style>
        .selection textarea::selection {
          background-color: black;
        }
      </style>
      <p>Select some text.</p>
      <form>
        <textarea id="selection">The quick brown fox jumps over the lazy dog.</textarea>
        <label for="selection">textarea</label>
      </form>
    </section>
```

**Comment by WilcoFiers** [comment]
I don't think this is a WCAG requirement. But as a best practice it makes sense. This seems like a pretty unusual edge case though. Do you have a real-world example of this?

**Comment by AutoSponge** [comment]
[This article](https://css-tricks.com/almanac/selectors/s/selection/) seems to suggest that changing the text color from #333 (9.19) to #8e44ad (4.27) is only an aesthetic choice. I'd argue that it breaks native, accessible, behavior.

**Comment by Malvoz** [comment]
I believe it makes sense to fold this into #975.
